docs/apache-airflow/upgrade-check.rst

```diff
@@ -0,0 +1,145 @@
+    Legacy UI is deprecated by default.......................................................SUCCESS
+    Logging configuration has been moved to new section......................................FAIL
+    Removal of Mesos Executor................................................................SUCCESS
+    Users must set a kubernetes.pod_template_file value......................................FAIL
+    SendGrid email uses old airflow.contrib module...........................................SUCCESS
+    Changes in import path of remote task handlers...........................................SUCCESS
+    Jinja Template Variables cannot be undefined.............................................FAIL
+    Found 7 problems.
+
+    ========================================== RECOMMENDATIONS ========================================
+
+    Fernet is enabled by default
+    ----------------------------
+    The fernet mechanism is enabled by default to increase the security of the default installation.
+
+    Problems:
+
+    1.  fernet_key in airflow.cfg must be explicitly set empty as fernet mechanism is enabledby default. This means that the apache-airflow[crypto] extra-packages are always installed.However, this requires that your operating system has libffi-dev installed.
+
+    Changes in import paths of hooks, operators, sensors and others
+    ---------------------------------------------------------------
+    Many hooks, operators and other classes has been renamed and moved. Those changes were part of unifying names and imports paths as described in AIP-21.
+    The contrib folder has been replaced by providers directory and packages:
+    https://github.com/apache/airflow#backport-packages
+
+    Problems:
+
+    1.  Using airflow.operators.python_operator.PythonOperator will be replaced by airflow.operators.python.PythonOperator. Affected file:
+
+
+The following sections describe what is being done and how to apply the recommendations shown above.
+Please note that the above results shown are only a partial set, where only the first
+two of the seven problems identified are shown in the section above. In reality,
+all the problems are shown on the screen.
+
+
+Understanding what is being checked
+'''''''''''''''''''''''''''''''''''
+
+The Upgrade Check checks the configuration data from airflow.cfg, the meta data from the Airflow
+database, as well as the DAGs which have been set up in the current Airflow environment.
+
+Using the above results as an example, there are two specific problems which have
+been identified.
+
+The first problem is identified in the configuration file airflow.cfg where the current configuration
+option for the fernet_key is no longer acceptable and needs to be changed.
+
+The second problem was identified in one of the DAGs. In this case, this import
+statement for the PythonOperator needs to be changed, since the location is different
+in Airflow 2.0. We will discuss how in the next section.
```

Comment:
```suggestion
Understanding what is being checked
'''''''''''''''''''''''''''''''''''

The Upgrade Check checks the configuration data from airflow.cfg, the metadata from the Airflow
database, as well as the DAGs which have been set up in the current Airflow environment.

Using the above results as an example, there are two specific problems which have
been identified.

The first problem is identified in the configuration file airflow.cfg where the current configuration
option for the fernet_key is no longer acceptable and needs to be changed.

The second problem was identified in one of the DAGs. In this case, this import
statement for the PythonOperator needs to be changed, since the location is different
in Airflow 2.0. We will discuss how in the next section.
```